File: winforms/aaudiocore/Strategies/RepeatStrategy.cs
using AAudioCore.Contracts;
using AAudioCore.Models;

namespace AAudioCore.Strategies;

public sealed class RepeatStrategy : IPlaybackStrategy
{
    public string Name => "repeat";

    public int SelectNext(int currentIndex, IReadOnlyList<PlaybackTrack> queue)
    {
        if (queue.Count == 0) return -1;
        return Math.Max(0, Math.Min(currentIndex, queue.Count - 1));
    }
}

File: react/aaudiocore/src/strategies/repeat-strategy.ts
import type { IPlaybackStrategy } from "../contracts/playback-strategy";
import type { PlaybackTrack } from "../models/track";

export class RepeatStrategy implements IPlaybackStrategy {
  readonly name = "repeat";

  selectNext(currentIndex: number, queue: PlaybackTrack[]): number {
    if (queue.length === 0) return -1;
    return Math.max(0, Math.min(currentIndex, queue.length - 1));
  }
}
